week3/841_keys_and_rooms.py

```diff
@@ -0,0 +1,17 @@
+# https://leetcode.com/problems/keys-and-rooms/
+from typing import List
+
+
+def can_visit_all_rooms(rooms: List[List[int]]) -> bool:
+    room_keys = {r: k for r, k in enumerate(rooms)}
+    visited_rooms = set()
+
+    def open_room(room):
```

Comment:
Nice and simple. Would have been a bonus points if you have a non-recursive version.